quasar_entities/src/event.rs

```diff
@@ -0,0 +1,119 @@
+//! `SeaORM` Entity. Generated by sea-orm-codegen 0.12.3
+use async_graphql::ComplexObject;
+use sea_orm::{entity::prelude::*, ActiveValue, Set};
+use serde_json::json;
+use stellar_xdr::{ContractEvent, Error, ScVal, WriteXdr};
+
+#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq, async_graphql::SimpleObject)]
+#[sea_orm(table_name = "events")]
+#[graphql(complex)]
+#[graphql(name = "Events")]
+pub struct Model {
+    #[sea_orm(primary_key)]
+    pub id: i32,
+    pub topic: String,
+    pub contract_id: String,
+    pub transaction_id: String,
+    pub value: Json,
+    pub r#type: String,
+}
+
+#[ComplexObject]
+impl Model {}
+
+#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
+pub enum Relation {}
+
+impl ActiveModelBehavior for ActiveModel {}
+
+impl TryFrom<ContractEvent> for ActiveModel {
+    type Error = Error;
+
+    fn try_from(event: ContractEvent) -> Result<Self, Self::Error> {
+        println!("{event:?}");
+        let (topic, value) = match &event.body {
+            stellar_xdr::ContractEventBody::V0(body) => {
+                let topic = &body.topics[0];
+                let topic = match topic {
+                    ScVal::Symbol(topic) => topic.to_string().unwrap(),
+                    _ => {
+                        panic!("not a symbol")
```

Comment:
This means that our app could panic if the data format from the Stellar node database is invalid, right? This could be for example because of a bug in the node itself. I think Quasar should be handle those types of error and it'd be better if we just returned an `Err` here.

What do you think?

Reply:
Resolved with https://github.com/eigerco/quasar/pull/43/commits/97627f34fa9fa1f052a0071ffd4fe7b4431b6519